action.php

```diff
@@ -1410,45 +1525,50 @@ protected function _render($raw) {
     /**
      * Finds out whether there is a discussion section for the current page
      *
-     * @param string $title
-     * @return bool
+     * @param string $title set to title from metadata or empty string
+     * @return bool discussion section is shown?
      */
-    protected function _hasDiscussion(&$title) {
+    protected function hasDiscussion(&$title)
+    {
         global $ID;
 
-        $cfile = metaFN($ID, '.comments');
+        $file = metaFN($ID, '.comments');
 
-        if (!@file_exists($cfile)) {
+        if (!@file_exists($file)) {
             if ($this->isDiscussionEnabled()) {
                 return true;
             } else {
                 return false;
             }
         }
 
-        $comments = unserialize(io_readFile($cfile, false));
+        $data = unserialize(io_readFile($file, false));
+
+        $title = $data['title'] ?? '';
 
-        if ($comments['title']) {
-            $title = hsc($comments['title']);
+        $num = $data['number'];
```

Comment:
seeing a warning for this line: `PHP message: PHP Warning:  Undefined array key "number" in /dokuwiki/lib/plugins/discussion/action.php on line 1549`

Reply:
Do I understand correctly that you got this warning when using this pull request?

Reply:
yep; on a page that does not have the discussion form showing, but has parts included through blog plugin using `{{blog>blog?2&firstseconly&nofooter&noeditbtn&nocomments}}`; the blog posts have discussion enabled: https://wild-water.nl/dokuwiki/blog/2015/0415_logboek_een_jaar_wildwatertochten
essentially this page: https://wild-water.nl/dokuwiki/start - but on my laptop

Reply:
You do not see at other places this warning?
I'm wondering if older version of this plugin might have not set this counter..
This one is fixed.

Reply:
👍🏽 I updated the branch, deleted the cache and ran though any pages with the discussion tag an nog longer see this popping up